synapse/lib/iq.py

```diff
@@ -528,6 +531,44 @@ def addTstForms(core):
         core.addDataModel('tst', modl)
         core.addTufoProp('inet:fqdn', 'inctest', ptype='int', defval=0)
 
+    @contextlib.contextmanager
+    def getAxonCore(self):
+        with self.getTestDir() as dirn:
+            neurconf = {'host': 'localhost', 'bind': '127.0.0.1', 'port': 0}
+            neurpath = s_common.gendir(dirn, 'neuron')
+            neur = s_neuron.Neuron(neurpath, neurconf)
+
+            blobpath = s_common.gendir(dirn, 'blob')
+            blobconf = {'host': 'localhost', 'bind': '127.0.0.1', 'port': 0}
+            blobauth = neur.genCellAuth('blob')
+            s_msgpack.dumpfile(blobauth, os.path.join(blobpath, 'cell.auth'))
+            blob = s_axon.BlobCell(blobpath, blobconf)
+            self.true(blob.cellpool.neurwait(timeout=3))
+
+            axonpath = s_common.gendir(dirn, 'axon')
+            axonauth = neur.genCellAuth('axon')
+            s_msgpack.dumpfile(axonauth, os.path.join(axonpath, 'cell.auth'))
+            axonconf = {'host': 'localhost', 'bind': '127.0.0.1', 'port': 0, 'axon:blobs': ('blob@localhost',)}
+            axon = s_axon.AxonCell(axonpath, axonconf)
+            self.true(axon.cellpool.neurwait(timeout=3))
+
+            core = s_cortex.openurl('ram:///')
+            self.addTstForms(core)
+
+            with TstEnv() as env:
+                env.axon = axon
```

Comment:
Use the ``.add(obj, fini=True)`` api on TstEnv to add the eventbuses.  Then you can just fini() the testenv object and it will fini all the buses for you in the order they were added.